public/js/hls-viewer.js

```diff
@@ -0,0 +1,169 @@
+      banner.classList.remove('hidden');
+      videoContainer.classList.add('hidden');
+      playVideoContainer.classList.add('hidden');
+    }
+  };
+
+  /**
+   * Listen for events on the OpenTok session
+   */
+  const setEventListeners = function (session) {
+    const streams = [];
+    const subscribers = [];
+    let broadcastActive = false;
+
+    /** Listen for a broadcast status update from the host */
+
+    session.on('signal:broadcast-url', function (event) {
+      console.log('signal:broadcast-url', event);
+      const broadcastUrl = event.data;
+      var video = document.getElementById('video');
+      if (Hls.isSupported()) {
+        var hls = new Hls();
+        console.log('signal:broadcast-url - ', broadcastUrl);
+        hls.loadSource(broadcastUrl);
+        hls.attachMedia(video);
+        hls.on(Hls.Events.MANIFEST_PARSED, function () {
+          playVideo();
+        });
+        hls.on(Hls.Events.BUFFER_EOS, function () {
+          updateBanner('ended');
+        });
+      } else if (video.canPlayType('application/vnd.apple.mpegurl')) {
+        video.src = broadcastUrl;
+        video.addEventListener('canplay', function () {
+          video.play();
+        });
+      }
+      updateBanner('active');
+    });
+  };
+
+  const playVideo = function () {
+    var video = document.getElementById('video');
+    if (video) {
+      video
+        .play()
+        .then((res) => {
+          console.log('Play Video Successfull', res);
+          const playVideoContainer = document.getElementById(
+            'play-video-container'
+          );
+          playVideoContainer.classList.add('hidden');
+        })
+        .catch((err) => {
+          console.log('Play Video error', err);
+        });
+    }
+  };
+
+  const switchToLiveMode = function () {
+    window.location.href = `viewer${window.location.search}`;
+  };
+
+  const addClickEventListeners = function () {
+    document.getElementById('play-video').addEventListener('click', playVideo);
+    document
+      .getElementById('go-live-btn')
+      .addEventListener('click', switchToLiveMode);
+  };
+
+  const init = function () {
+    addClickEventListeners();
+    getBroadcastUrl()
+      .then((data) => {
+        updateBanner('active');
+
+        const { url } = data;
+
+        var video = document.getElementById('video');
+        if (Hls.isSupported()) {
+          var hls = new Hls();
+          console.log('broadcast-url - ', url);
+          hls.loadSource(url);
+          hls.attachMedia(video);
+          hls.on(Hls.Events.MANIFEST_PARSED, function () {
+            playVideo();
+          });
+        } else if (video.canPlayType('application/vnd.apple.mpegurl')) {
+          video.src = url;
+          video.addEventListener('canplay', function () {
+            video.play();
+          });
+        }
+      })
+      .catch((e) => console.log(e)); // addClickEventListeners();
+    // const credentials = getCredentials();
```

Comment:
intentional?

Reply:
yes, forgot to remove the comments. Thanks